### PR TITLE
docs(notifications): split to multiple docs and add more info

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

- split docs to 3 different pages; getting started, processors and usage
- add information about email notifications
- add information about scaffolder module

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by drodil** [comment]
Not stale

File: docs/notifications/processors.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      port: 587
+      secure: false
+      username: ${SMTP_USERNAME}
+      password: ${SMTP_PASSWORD}
+```
+
+Apart from STMP, the email processor also supports the following transmissions:
+
+- SES
+- sendmail
```
Comment (benjdlambert) [inline docs/notifications/processors.md:104]: Might need to add this to the `vocab` in `.github` folder

File: docs/notifications/processors.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+---
+id: processors
+title: Processors
+description: How to setup notification processors
+---
+
+The notifications can be extended with `NotificationProcessor`. These processors allow to decorate notifications before they are sent or/and send the notifications to external services.
+
+Depending on the needs, a processor can modify the content of a notification or route it to different systems like email, Slack, or other services.
```
Comment (aramissennyeydd) [inline docs/notifications/processors.md:9]:
```suggestion
Depending on your needs, a processor can modify the content of a notification or route it to different systems like email, Slack, or other services.
```

File: docs/notifications/processors.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
+---
+id: processors
+title: Processors
+description: How to setup notification processors
+---
+
+The notifications can be extended with `NotificationProcessor`. These processors allow to decorate notifications before they are sent or/and send the notifications to external services.
```
Comment (aramissennyeydd) [inline docs/notifications/processors.md:7]:
```suggestion 
Notifications can be extended with `NotificationProcessor`. These processors allow you to decorate notifications before they are sent and/or send the notifications to external services.
```

File: docs/notifications/processors.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      from: 'backstage',
+      to: 'user',
+      subject: notification.payload.title,
+      text: notification.payload.description,
+    });
+  }
+}
+```
+
+Both of the processing functions are optional, and you can implement only one of them.
```
Comment (aramissennyeydd) [inline docs/notifications/processors.md:42]:
```suggestion
Both of the processing functions are optional, and you can just implement one of them.
```

File: docs/notifications/processors.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        notifications.addProcessor(new MyNotificationProcessor());
+      },
+    });
+  },
+});
+```
+
+## Built-in Processors
+
+Backstage comes with some built-in processors that can be used out of the box.
```
Comment (aramissennyeydd) [inline docs/notifications/processors.md:69]:
```suggestion
Backstage comes with some processors that can be used immediately.
```

File: docs/notifications/usage.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+export const myPlugin = createBackendPlugin({
+  pluginId: 'myPlugin',
+  register(env) {
+    env.registerInit({
+      deps: {
+        // ...
+        notificationService: notificationService,
+      },
+      async init({ config, logger, httpRouter, notificationService }) {
```
Comment (aramissennyeydd) [inline docs/notifications/usage.md:49]:
```suggestion
      async init({ 
        // ...
      	notificationService 
     }) {
```

File: docs/notifications/usage.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+## Signals
+
+The use of signals with notifications is optional but generally enhances user experience and performance.
+
+When a notification is created, a new signal is emitted to a general-purpose message bus to announce it to subscribed listeners.
+
+The frontend maintains a persistent connection (WebSocket) to receive these announcements from the notifications channel.
+The specific details of the updated or created notification should be retrieved via a request to the notifications API, except for new notifications, where the payload is included in the signal for performance reasons.
+
+In a frontend plugin, to subscribe for notifications' signals:
```
Comment (aramissennyeydd) [inline docs/notifications/usage.md:130]:
```suggestion
In a frontend plugin, to subscribe to notifications' signals:
```

**Review by aramissennyeydd — APPROVED** [review]
Thanks for this! Looks good, had a few small suggestions as I was reading through it.

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
@drodil could you maybe rebase here to see if we can fix the microsite build? :pray:

**Comment by drodil** [comment]
Will do as soon as I beat this 40 degree fever 🤒

**Comment by pjungermann** [comment]
oh... it seems I had a bad/wrong reference. 😮  Definitely not intended.
Thanks @benjdlambert

**Comment by drodil** [comment]
Not stale

**Comment by awanlin** [comment]
Hi @drodil, I was going to merge this as it's been triple approved but seems the CI is not passing, prettier is unhappy. Can you give it a look, please?

**Comment by drodil** [comment]
@awanlin fix'd! ✅

**Comment by awanlin** [comment]
Sorry, I was too slow with this and conflicts got in again :(

**Comment by drodil** [comment]
@awanlin new try with rebase (hope I got it right... 😆 )

**Review by awanlin — DISMISSED** [review]
Didn't notice this till now but we need a changeset for that `README` change so that it gets updated on NPM too. The rest is good though!

**Review by benjdlambert — APPROVED** [review]
Nice! looks good! let's :ship:

**Pull request merged**